src/main/java/bio/terra/service/dataset/flight/ingest/IngestValidateAzureRefsStep.java

```diff
@@ -0,0 +1,77 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.common.Column;
+import bio.terra.common.Table;
+import bio.terra.model.BillingProfileModel;
+import bio.terra.model.TableDataType;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.azure.AzureSynapsePdao;
+import bio.terra.service.filedata.azure.tables.TableDirectoryDao;
+import bio.terra.service.profile.flight.ProfileMapKeys;
+import bio.terra.service.resourcemanagement.ResourceService;
+import bio.terra.service.resourcemanagement.azure.AzureAuthService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class IngestValidateAzureRefsStep extends IngestValidateRefsStep {
+
+  private final AzureAuthService azureAuthService;
+  private final ResourceService resourceService;
+  private final DatasetService datasetService;
+  private final AzureSynapsePdao azureSynapsePdao;
+  private final TableDirectoryDao tableDirectoryDao;
+
+  public IngestValidateAzureRefsStep(
+      ResourceService resourceService,
+      AzureAuthService azureAuthService,
+      DatasetService datasetService,
+      AzureSynapsePdao azureSynapsePdao,
+      TableDirectoryDao tableDirectoryDao) {
+    this.azureAuthService = azureAuthService;
+    this.resourceService = resourceService;
+    this.datasetService = datasetService;
+    this.azureSynapsePdao = azureSynapsePdao;
+    this.tableDirectoryDao = tableDirectoryDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    var workingMap = context.getWorkingMap();
+    var flightId = context.getFlightId();
+    var dataset = IngestUtils.getDataset(context, datasetService);
+
+    var billingProfile = workingMap.get(ProfileMapKeys.PROFILE_MODEL, BillingProfileModel.class);
+    var storageAccountResource =
+        resourceService.getOrCreateStorageAccount(dataset, billingProfile, flightId);
```

Comment:
It looks like we already run getOrCreateStorageAccount() in one of the previous step (IngestCreateTargetDataSourceStep). IMO, it's better to pull values out of the flight map rather than retrieving the value again, especially when there's a chance things get out of sync (the "create" option for getOrCreateStorageAccount()). The pattern of putting/pulling the storage account out of the flight map is also the pattern we follow in the[ file ingest flights](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileAzureMakeStorageAccountLinkStep.java#L51). 

(So, this is a two step change - put the storage account into the map in IngestCreateTargetDataSourceStep and pull it out in this step)

Reply:
Good point. Stuck it in the working map.